source/jormungandr/jormungandr/scenarios/helper_classes/helper_utils.py

```diff
@@ -391,6 +415,193 @@ def add_poi_access_point_in_sections(fallback_type, via_poi_access, sections):
+
+    Args:
+        cord1 (tuple): The (latitude, longitude) of the starting point.
+        cord2 (tuple): The (latitude, longitude) of the destination point.
+        speed_kmh (float, optional): Walking speed in kilometers per hour. Defaults to 5 km/h.
+
+    Returns:
+        float: The walking time in secondes.
+    """
+    distance = haversine(cord1, cord2)
+    # Time in hours
+    time_hours = distance / speed_kmh
+    # Convert to secondes
+    seconds = time_hours * 60 * 60
+    return round(seconds)
+
+
+def _get_place(kwargs, uri):
+    """
+    Retrieve a place instance based on the provided URI.
+
+    Args:
+        kwargs (dict): A dictionary containing the following keys:
+            - future_manager: The future manager instance.
+            - instance: The instance to be used.
+            - request_id: The ID of the request.
+        uri (str): The URI of the place to retrieve.
+
+    Returns:
+        PlaceByUri: An instance of PlaceByUri after waiting for the result.
+    """
+    place_by_uri_instance = PlaceByUri(kwargs["future_manager"], kwargs["instance"], uri, kwargs["request_id"])
+    return place_by_uri_instance.wait_and_get()
+
+
+def _update_fallback_with_bike_mode(
+    journey, fallback_dp, fallback_period_extremity, fallback_type, via_pt_access, via_poi_access, **kwargs
+):
+    """
+    Updates the journey with bike mode fallback sections.
+
+    This function updates the journey sections with bike mode fallback sections based on the fallback type
+    (BEGINNING_FALLBACK or ENDING_FALLBACK). It aligns the fallback direct path datetime, updates the section IDs,
+    and creates the necessary links between the fallback and public transport parts. It also handles the addition
+    of POI access points in the sections.
+
+    Args:
+        journey (Journey): The journey object to be updated.
+        fallback_dp (DirectPath): The direct path object for the fallback.
+        fallback_period_extremity (datetime): The extremity datetime for the fallback period.
+        fallback_type (StreetNetworkPathType): The type of fallback (BEGINNING_FALLBACK or ENDING_FALLBACK).
+        via_pt_access (PtObject): The public transport access point object.
+        via_poi_access (POIObject): The point of interest access point object.
+        **kwargs: Additional keyword arguments, including:
+            - origin_mode (list): The mode of origin (e.g., ["bike"]).
+            - destination_mode (list): The mode of destination (e.g., ["bike"]).
+            - future_manager (FutureManager): The future manager instance.
+            - instance (Instance): The instance object.
+            - request_id (str): The request ID.
+            - additional_time (timedelta): The additional time to be added to the sections.
+
+    Returns:
+        None
+    """
+    # Validate required arguments
+    if not all(kwargs.get(key) for key in ("future_manager", "instance", "request_id")):
```

Comment:
if keys are mandatory, you may as well directly put them in function's signature